buildSrc/src/main/kotlin/VersionTuple.kt

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import java.nio.file.Files
+import java.nio.file.Path
+import java.util.regex.Pattern
+
+/** Represents a version tuple with mandatory major, minor and patch numbers and snapshot-flag. */
+data class VersionTuple(val major: Int, val minor: Int, val patch: Int, val snapshot: Boolean) :
+  Comparable<VersionTuple> {
+
+  companion object Factory {
+    val pattern =
+      Pattern.compile(
+        "^(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?\$"
+      )
+
+    fun fromFile(file: Path): VersionTuple = create(Files.readString(file, Charsets.UTF_8).trim())
```

Comment:
nit: method is overloaded and defaults to UTF8 already (same for `writeString` below)